flood_api/routers/flood.py

```diff
@@ -6,132 +6,182 @@
     ThresholdDataDep,
 )
 from flood_api.dependencies.queryparams import (
+    BoundingBoxDep,
     CoordinatesDep,
     DateRangeDep,
     IncludeNeighborsDep,
 )
 from flood_api.models.detailed_types import DetailedProperties, DetailedResponseModel
 from flood_api.models.summary_types import SummaryProperties, SummaryResponseModel
 from flood_api.models.threshold_types import ThresholdProperties, ThresholdResponseModel
-from flood_api.utils.geospatial_operations import get_data_for_point
+from flood_api.utils.geospatial_operations import get_data_for_bbox, get_data_for_point
 from flood_api.utils.json_utilities import dataframe_to_geojson
-from flood_api.utils.validation_helpers import validate_coordinates, validate_dates
+from flood_api.utils.validation_helpers import (
+    validate_bounding_box,
+    validate_coordinates,
+    validate_dates,
+    validate_generic_inputs,
+)
 
 router = APIRouter(tags=["flood"])
 
 
 @router.get(
     "/summary",
-    summary="Get summary forecast",
-    description="Returns a summary forecast of the next 30 days for the given location",
+    summary="Get summary forecast for a location",
+    description="Returns a summary forecast of the next 30 days either for the cell at the given coordinates or for the cells within the given bounding box",
 )
 async def summary(
     gdf: SummaryDataDep,
     coordinates: CoordinatesDep,
+    bbox: BoundingBoxDep,
     include_neighbors: IncludeNeighborsDep,
 ) -> SummaryResponseModel:
-    longitude, latitude = coordinates
-
-    # Validate the inputs
-    validate_coordinates(latitude=latitude, longitude=longitude)
-
-    queried_cell, neighboring_cells = get_data_for_point(
-        latitude=latitude,
-        longitude=longitude,
-        include_neighbors=include_neighbors,
-        gdf=gdf,
+    validate_generic_inputs(
+        coordinates,
+        bbox,
+        possible_inputs_str="coordinates, bounding box",
     )
 
+    if coordinates is not None:
+        validate_coordinates(*coordinates)
```

Comment:
Could be challenging because there first needs to be a check that either `coordinates` or `bbox` is provided. So if the user submits a `bbox`, both `lat` and `lon` in `coordinates` will be `None`.